src/state-manager/TransactionConsensus.ts

```diff
@@ -1220,6 +1238,31 @@ class TransactionConsenus {
     )
   }
 
+  verifyTwoThirdsMajority(receipt: AppliedReceipt2, executionGroupNodes: Set<string>): boolean {
+    const uniqueSignOwners = new Set(receipt.signatures.map(sign => sign.owner));
+    const totalNodes = executionGroupNodes.size;
+    const requiredMajority = Math.ceil((2 / 3) * totalNodes);
+    if (uniqueSignOwners.size < requiredMajority) {
+      return false;
+    }
+
+    const vote = receipt.appliedVote; 
+    const voteHash = this.calculateVoteHash(vote);
+    const appliedVoteHash = {
+      txid: vote.txid,
+      voteHash,
+    }
+
+    let validSignatures = 0;    
+    for (const owner of uniqueSignOwners) {
+      const signedObject = { ...appliedVoteHash, sign: receipt.signatures.find(sign => sign.owner === owner) };
```

Comment:
Can we use a map from owner to sign? Since this code is going to be a hot path it will be more efficient